sidecar/src/watchdog.rs

```diff
@@ -46,12 +46,44 @@ impl Watchdog {
         let mem_usage_bytes = Arc::new(AtomicUsize::new(0));
         let handle_mem_usage_bytes = mem_usage_bytes.clone();
 
+        let still_alive = Arc::new(AtomicU32::new(0));
+        let still_alive_thread = still_alive.clone();
+
+        let interval = self.interval.period();
+        std::thread::spawn(move || {
+            let mut repeats = false;
+            let mut last = 0;
+            loop {
+                std::thread::sleep(interval);
+                let current = still_alive_thread.load(Ordering::Relaxed);
+                if last != current {
+                    if current == u32::MAX {
+                        return;
+                    }
+                    last = current;
+                    repeats = false;
+                } else {
+                    if repeats {
+                        std::thread::spawn(move || {
```

Comment:
Should we have a telemety metric as well?

Reply:
If it's stuck, it'll be unable to flush that one though. So emitting it would be non-trivial...

Reply:
Ok. Thanks